ganga/GangaDirac/Lib/Server/DiracProcessManager.py

```diff
@@ -0,0 +1,117 @@
+import os
+import uuid
+import psutil
+
+from aioprocessing import AioManager, AioQueue
+from GangaDirac.Lib.Utilities.DiracUtilities import GangaDiracError, getDiracEnv
+from GangaCore.GPIDev.Credentials import credential_store
+from GangaCore.Utility.logging import getLogger
+
+from .DiracExecutorProcess import DiracProcess
+
+
+logger = getLogger()
+
+
+class Singleton(type):
+    _instances = {}
+
+    def __call__(cls, *args, **kwargs):
+        if cls not in cls._instances:
+            cls._instances[cls] = super(Singleton, cls).__call__(*args, **kwargs)
+        return cls._instances[cls]
+
+
+class AsyncDiracManager(metaclass=Singleton):
+    def __init__(self):
+        self.dirac_process = None
+        self.manager = None
+        self.task_queues = {}
+        self.task_result_dicts = {}
+        self.active_processes = {}
+        self.task_queue = AioQueue()
+
+    def prepare_process_env(self, env=None, cred_req=None):
+        if env is None:
+            if cred_req is None:
+                env = getDiracEnv()
+            else:
+                env = getDiracEnv(cred_req.dirac_env)
+        if cred_req is not None:
+            env['X509_USER_PROXY'] = credential_store[cred_req].location
+            if os.getenv('KRB5CCNAME'):
+                env['KRB5CCNAME'] = os.getenv('KRB5CCNAME')
+
```

Comment:
I wonder if this does the right thing. Testing from another user seems to imply that we do not take the value of `X509_USER_PROXY` correctly into account.

Reply:
Yes, I did ask for someone with different credentials to do a test for that since I couldn't test it with a different set of creds myself. I am basically doing exactly [what the old process did](https://github.com/ganga-devs/ganga/blob/0ceb3ebba7699ae22564a855560971213417af11/ganga/GangaDirac/Lib/Utilities/DiracUtilities.py#L277).

Now, I believe that `X509_USER_PROXY` is only read by [DiracProxy.py](https://github.com/ganga-devs/ganga/blob/0ceb3ebba7699ae22564a855560971213417af11/ganga/GangaDirac/Lib/Credentials/DiracProxy.py#L64) which has remained untouched

Reply:
Is there some kind of error to indicate an issue with the grid proxy which might provide some more info?